### DISCUSSION TITLE
Is it required to have initial net momentum = 0 when using periodic boundary condition?

### DISCUSSION THREAD

**Discussion by cooluser26** [epochpic/epoch#740] [Q&A]
In the 1D two-stream example deck, I notice that the electrons are set up in a way such that the initial system's momentum is zero. Is this a requirement for periodic B.Cs simulations? I'm asking because I observed a peculiar behaviour when the initial momentum isn't zero:

- Here is the field energy plot when the initial net momentum = 0. This is all pretty reasonable
<img width="283" alt="image" src="https://github.com/user-attachments/assets/95da771b-0d5c-472a-a30f-1a39a247e96f">

- Here is the same plot but when the initial net momentum is **not** zero (i.e stationary background electrons and drifting "beam" electrons). This seems to be some **numerical artefact**?  Why does this happen? 
![image](https://github.com/user-attachments/assets/1f783102-8cfe-4c13-9d36-91dec93b60c7)

Below is my input deck for the electron two-stream instability. Note that I also compiled the code with fifth order particle shape.

[two_stream.txt](https://github.com/user-attachments/files/17670525/two_stream.txt)

Thank you for any help/insight you can offer!

**Comment by Status-Mirror** [comment]
Hi @wnguyen1312 

I think the two streams initially pass through each other as this is the setup for the two-stream instability problem. Starting a species at rest is a different physical problem, so the evolution would be different.

If you initially have a dense beam moving in one direction, your system starts with a net current density. This will create fields which may oppose the motion. These waves you're seeing might just be plasma oscillations. If you look at the total particle energy AND the total field energy with time, the system energy may be better conserved.

Hope this helps,
Stuart
